src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java

```
@@ -544,15 +547,158 @@ public boolean isPluginEnabled(Plugin plugin)
 	public Plugin[] loadPlugins(File directory)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		try 
+		{
+			//copy to temp folder
+			File tempDir = createTemporaryDirectory("MockBukkit-plugins");
+			Path targetPath = Files.copy(directory.toPath(), tempDir.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+			Map<String, File> pluginsToLoad = new HashMap<>();
+			Map<String, Set<String>> softDependencies = new HashMap<>();
+			Map<String, Set<String>> dependencies = new HashMap<>();
+
+			for(File file : targetPath.toFile().listFiles())
+			{
+				try 
+				{
+					//check description validity
+					PluginDescriptionFile pdf = loader.getPluginDescription(file);
+
+					//acknowledge plugin name restrictions
+					String pluginName = pdf.getName();
+					checkPluginNameValidity(file, pluginName);
+
+					//check if already registered
+					File previousRegisteredPlugin = pluginsToLoad.put(pluginName, file);
+					if (previousRegisteredPlugin != null) 
+					{
+						throw new InvalidDescriptionException(String.format("Plugin %s wasn't loaded: Duplicate plugin name %s!", file.getName(), pluginName));
+					}
+
+					//gather softDependencies, dependencies, and loadBefore
+					Set<String> pluginSoftDependencies = softDependencies.getOrDefault(pluginName, new HashSet<>());
+					pluginSoftDependencies.addAll(pdf.getSoftDepend());
+					softDependencies.put(pluginName, pluginSoftDependencies);
+
+					Set<String> pluginDependencies = dependencies.getOrDefault(pluginName, new HashSet<>());
+					pluginDependencies.addAll(pdf.getDepend());
+					dependencies.put(pluginName, pluginDependencies);
+
+					for (String loadBefore : pdf.getLoadBefore())
+					{
+						Set<String> softDepends = softDependencies.getOrDefault(loadBefore, new HashSet<>());
+						softDepends.add(pluginName);
+						softDependencies.put(loadBefore, softDepends);
+					}
+				}
+				catch(InvalidDescriptionException e)
+				{
+					e.printStackTrace();
```

You might want to wrap and rethrow the exception here to make sure that the unit tests will fail properly, instead of just printing a stack trace and continuing. Many test runners hide any output unless the test actually fails.